### PR TITLE
Added profile type hexagon

### LINKED ISSUE #122: Provide a profile-type hexagon
**Summary**

In 3-roll processes sometimes in_profiles with a hexagonal cross-section a used.
PyRolL should provide a respective profile-type

### PR DESCRIPTION
**Pull request by ChRen95**
Closes #122.

### CODE REVIEW

File: pyroll/core/profile/profile.py [hunk 1]
```diff
@@ -622,3 +645,86 @@ def __attrs__(self):
...
+            corner_radius: float = 0,
+            **kwargs
+    ):
+        """
+        Creates a hexagonal shaped profile (a real hexagonal shape with rounded corners,
+        without imperfections of hexagonal grooves). A hexagon is oriented to stand on its side.
+        Give exactly one of ``side`` and ``height``.
+
+        :param side: the side length of the hexagonal profile, must be > 0
+        :param diagonal: the height of the hexagonal profile when standing on the flat base, must be > 0
```
Comment (axtimhaus) [inline pyroll/core/profile/profile.py:664]: May be better to define diagonal as the corner-to-corner distance and height as the side-to-side distance distinctly.

File: pyroll/core/rotator/hookimpls.py [hunk 2]
```diff
@@ -74,3 +74,9 @@ def upset_in_0(self: Rotator):
 def three_roll_pass_180(self: Rotator):
     if "3fold" in self.next_roll_pass.classifiers:
         return 180
+
+
+@Rotator.rotation
+def three_roll_pass_hexagon_to_hexagon(self: Rotator):
+    if "3fold" in self.next_roll_pass.classifiers and "flat" in self.next_roll_pass.classifiers:
+        return 0
```
Comment (axtimhaus) [inline pyroll/core/rotator/hookimpls.py:82]: Really zero? Not 180 like in default case?

**Review by axtimhaus — CHANGES_REQUESTED** [review]

File: pyroll/core/rotator/hookimpls.py (hunk 2 above)
Comment (ChRen95) [reply pyroll/core/rotator/hookimpls.py:82]: Just looked it up again is 180.

File: pyroll/core/profile/profile.py (hunk 1 above)
Comment (ChRen95) [reply pyroll/core/profile/profile.py:664]: so you would measure including the radius or where the projected sides would meet?

Comment (axtimhaus) [reply pyroll/core/profile/profile.py:664]: The side intersection, so this is coherent with squares.

**Review by axtimhaus — APPROVED** [review]

**Pull request merged**